context-propagation/src/test/java/io/smallrye/mutiny/context/MultiContextPropagationTest.java

```diff
@@ -231,4 +229,63 @@ public void testCompletionStage() throws InterruptedException, ExecutionExceptio
         int result = cf.get();
         assertThat(result).isEqualTo(2);
     }
+
+    @Test
+    public void testOnSubscribe() {
+        MyContext ctx = MyContext.get();
+        assertThat(ctx).isNotNull();
+
+        AssertSubscriber<Long> subscriber = Multi.createFrom().ticks().every(Duration.ofMillis(1))
+                .select().first(5)
+                .onSubscribe().invoke(() -> {
+                    assertThat(ctx).isEqualTo(MyContext.get());
```

Comment:
Instead of `isEqualTo` checks here and in the remainder I'd use `isSameAs` because it's really what you want to check.

Reply:
yep